src/main/scala/piconot/parser/Parser.scala

```diff
@@ -0,0 +1,81 @@
+package piconot.parser
+
+import scala.util.parsing.combinator._
+import picolib.semantics._
+/**
+ * -----------
+ * Grammar
+ * -----------
+ * 
+ *          r Rule ::= st sr* -> c (st)
+ *          st State ::= Int
+ *          c Command ::= N | E | S | W | X
+ *          sr Surroundings ::== No (sr) | Nx (sr) | So (sr) ...
```

Comment:
Writing this grammar was actually really key for organizing our thoughts so we could start coding.